src/nvlink/constants.py

```
@@ -0,0 +1,223 @@
+        Key in 16-digit format (e.g., "2025121544030508")
+
+    Raises:
+        ValueError: If parameters are invalid
+
+    Examples:
+        >>> generate_nar_time_series_full_key("20251215", "44", 3, 5, 8)
+        '2025121544030508'
+    """
+    # Validate date format
+    if not isinstance(date, str) or len(date) != 8 or not date.isdigit():
+        raise ValueError(f"Invalid date format: {date}. Must be YYYYMMDD format.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)